pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -323,24 +325,50 @@ func (m *kubeGenericRuntimeManager) generateContainerConfig(ctx context.Context,
 	}
 	containerLogsPath := buildContainerLogsPath(container.Name, restartCount)
 	restartCountUint32 := uint32(restartCount)
+
+	parsedImageName, _, _, err := parsers.ParseImageName(container.Image)
+	if err != nil {
+		return nil, cleanupAction, fmt.Errorf("parse image name: %w", err)
+	}
+
+	keyring, err := credentialprovidersecrets.MakeDockerKeyring(pullSecrets, m.keyring)
+	if err != nil {
+		return nil, cleanupAction, fmt.Errorf("make keyring: %w", err)
+	}
+
+	creds, _ := keyring.Lookup(parsedImageName)
+	auths := []*runtimeapi.AuthConfig{}
```

Comment:
If we are adding a new API to include credentials, I think that API should provide _all_ the available credentials, to support mirror registries that require credentials.

OTOH that would only really _work_ if `PullImage` also provided all the available credentials, which is not currently the case. So doing that here would not be currently _effective_, the only benefit of doing that would be so that if/when `PullImage` is extended, this `ContainerConfig` filed will not see a value change.

Reply:
I would keep the scope of this PR to extend the `ContainerConfig`. The credentials extension is tracked in a enhancement https://github.com/kubernetes/enhancements/issues/3060.